front-end/src/components/AnalysisPage.test.js

```diff
@@ -0,0 +1,22 @@
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import { Provider } from 'react-redux';
+import store from '../store.js';
+import AnalysisPage from './AnalysisPage.jsx';
+
+describe('AnalysisPage tests', () => {
+    const renderComponent = () => render(
+        <Provider store={store} >
+            <AnalysisPage />
+        </Provider>
+    );
+
+    it('should dispatch openMainPage when Return to Main Page button is clicked', () => {
+        renderComponent();
+        const spy = jest.spyOn(store, 'dispatch');
```

Comment:
Better to name this dispatch IMO (that way the final line reads as dispatch being called) (otherwise this test is 👍 )